src/main/java/org/logstash/dissect/Delimiter.java

```diff
@@ -0,0 +1,42 @@
+package org.logstash.dissect;
+
+import org.logstash.dissect.search.DelimiterSearchStrategy;
+
+public final class Delimiter {
+    private final byte[] needle;
+    private final String delimiter;
+    private final int size;
+    private DelimiterSearchStrategy strategy;
+
+    public Delimiter(String delim) {
```

Comment:
Just as a hint: Should you plan to put `Delimiter` instances in collections like `HashSet` you should implement `#equals(Object)` and `#hashCode()` (no need to do it, you should just be aware of that).